lib/misc.py

```diff
@@ -36,7 +36,7 @@ def __call__(self, *args, **kwargs):
 def delete(cli):
    ch.dependencies_check()
    img_ref = ch.Image_Ref(cli.image_ref)
-   img = ch.Image(img_ref, cli.storage)
+   img = ch.Image(img_ref)
```

Comment:
So this was a little subtle. I think with your solution you added a second error that compensated for the first, so and it ended up working, i.e. it was a “two wrongs make a right” approach. ;)

The initial error was passing the storage directory when `Image` expected the complete path. If that's not given, then it computes the unpack path based on the storage directory. It worked with `--storage` omitted because in that case it gets `unpack_path=None`, which is the same as the default.